kernel/shared/src/main/scala/cats/effect/kernel/Resource.scala

```diff
@@ -1223,7 +1223,7 @@ private[effect] trait ResourceHOInstances0 extends ResourceHOInstances1 {
   implicit def catsEffectAsyncForResource[F[_]](implicit F0: Async[F]): Async[Resource[F, *]] =
     new ResourceAsync[F] {
       def F = F0
-      override def applicative = this
+      override def applicative: ResourceAsync[F] = this
```

Comment:
I'm not sure if it matters, but can this be:
```suggestion
      override def applicative: Async[Resource[F, *]] = this
```

Reply:
```
typelevel/cats-effect/kernel/shared/src/main/scala/cats/effect/kernel/Resource.scala:1226:20: incompatible type in overriding
[error] override def applicative: cats.effect.kernel.ResourceAsync[F] (defined in class ResourceAsync);
[error]  found   : cats.effect.kernel.Async[[β$26$]cats.effect.kernel.Resource[F,β$26$]]
[error]  required: cats.effect.kernel.ResourceAsync[F]
[error]       override def applicative: Async[Resource[F, *]] = this
[error]                    ^
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
```

🤔

Reply:
Thanks. I was surprised some of the other ones could use the more general type. 🤔